File: uniffi_bindgen/src/bindings/kotlin/templates/Float32Helper.kt
internal object FfiConverterFloat {
    fun lift(v: Float): Float {
        return v
    }

    fun read(buf: ByteBuffer): Float {
        return buf.getFloat()
    }

    fun lower(v: Float): Float {
        return v
    }

    fun write(v: Float, buf: RustBufferBuilder) {
        buf.putFloat(v)
    }
}

File: uniffi_bindgen/src/bindings/swift/templates/Float32Helper.swift
fileprivate struct FfiConverterFloat: FfiConverterPrimitive {
    typealias FfiType = Float
    typealias SwiftType = Float

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> Float {
        return try lift(readFloat(&buf))
    }

    static func write(_ value: Float, into buf: inout [UInt8]) {
        writeFloat(&buf, lower(value))
    }
}
